### PR TITLE
myb3k: WIP adding ISA bus, cards, devices and branded clones

### PR DESCRIPTION
**Pull request by JoakimLarsson**
This is work in progress based on Ericsson step/one Service Manuals, a branded Mybrain 3000

### CODE REVIEW

File: src/devices/bus/isa/myb3k_fdc.cpp [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+//-------------------------------------------------
+//  myb3k_fdc_command - descrete fdc card features
+//-------------------------------------------------
+#define FDC_MSM_MODE   0x40
+#define FDC_DDEN       0x20
+//#define FDC_MOTOR_ON   0x10 // According to service manual but not schematics and BIOS
+#define FDC_SIDE_SEL   0x08
+#define FDC_MOTOR_ON   0x04 // According to schematics but "Motor Cont" according to service manual
+#define FDC_DRIVE_SEL  0x03
```
Comment (cuavas) [inline src/devices/bus/isa/myb3k_fdc.cpp]: If these are only going to be used in one function, please use local `constexpr` or enum or something.  Preprocessor macros can't be scoped and produce unhelpful error messages when things go wrong, and hence should be avoided if something else will do the same job.

Comment (JoakimLarsson) [reply src/devices/bus/isa/myb3k_fdc.cpp]: ok, I'll add it to the upcoming update of Mybrain 3000

Comment (JoakimLarsson) [reply src/devices/bus/isa/myb3k_fdc.cpp]: By the way @cuavas , as you can see the classes for the two ISA card in this file are 99% the same code, only the fdc device type and the number of floppies differ. I could use some C++ hints how to create a base class with all the methods and only keep the last 1% in the derived classes, I tried but the exercise is giving me grave headaches. I will add the 8" fdc as well when I get hands on one and I would hate to tripple the code...

I figure the memory map, the fdc device and the floppy configuration will differ

**Comment by rb6502** [comment]
@JoakimLarsson look at the SoundBlaster card - it demonstrates deriving multiple ISA cards from a single base class.

**Comment by JoakimLarsson** [comment]
Thx, I'll do it when I find time to add the 8" fdc card. The main problem I had was to put all methods in the base class that operated on different WD devices for each card. Obviously I keep my fingers crossed that the 8" is also a WD device. I guess I have to downcast them to some common device type but never got through the ocean of errors in the excercise. The compiler wouldn't take my word for that the pointers were pointing to closely related objects

**Pull request merged**